docs/advanced-topics/client-anticipation.md

```diff
@@ -0,0 +1,117 @@
+
+:::
+
+## Smoothing
+
+Much of the time, when you perform an anticipation action, your anticipation will be correct and the server will update the state to the same value as you did. Occasionally, though, the server will have a different result, since latency can result in the server state changing before it receives your command in ways that affect the result. When this happens, by default, the value will be snapped to the new server value.
+
+In some cases, this may result in an undesirable player experience, causing things that normally move smoothly to become choppy. When this happens, you can smooth out the result by using the `Smooth()` function on both `AnticipatedNetworkVariable` and `AnticipatedNetworkTransform`.
+
+`Smooth()` takes a starting position and an ending position (which would usually be the prevoius anticipated value and either the new anticipated value or the current authoritatvie value) and a duration over which to perform the smoothing. For `AnticipatedNetworkVariable`, because the type is generic, it also requires a delegate to use to perform the smoothing operation (which takes a from value, a to value, and a percent value, and in many cases may just be `Mathf.Lerp`).
+
+Because the first parameter to `Smooth()` is usually the previous anticipated value, and by the time the `OnReanticipate` function is called, the anticipated value has been overwritten with the authoritative value, you can use `PreviousAnticipatedValue` on `AnticipatedNetworkVariable` and `PreviousAnticipatedState` on `AnticipatedNetworkTransform` to access the value you most recently passed into `Anticipate()` or its related transform methods. Note, however, that if you call `Anticipate()` or its related transform methods prior to doing the smoothing, they will update the previous anticipated value, so you may need to store the previous anticipated value in a local variable before doing your reanticipation.
+
+### Server-Side Smoothing
+
+Even though these concepts are mostly for client-side use, there are cases where you might want to perform server-side smoothing as well. For example, when running in host mode, the host player will see the movement of other players. Due to latency and jitter, input updates from remote clients may not arrive every frame, which may result in choppy movement of an AnticipatedNetworkTransform, for example.
+
+To address this, you can also use the `Smooth()` function at any time on the server side. The server does not do anticipation, so it will not get any `OnReanticipate()` events, but you can, for example, call `Smooth()` each time you receive an input from the client in order to smooth the motion between the previous position and the new one.
```

Comment:
Should that really be handled by Netcode? Couldn't users interpolate their own authoritative data or just use Physics interpolation for example?

Reply:
On the client side, yes, though it's such a common thing that I figured it would be useful to give users a method for it.

On the server side, smoothing is still used for interpolation, and it was a huge pain to implement it for AnticipatedNetworkTransform. They'd have a hard time implementing it themselves.